iree/compiler/Dialect/VMLA/Transforms/PreConversionLowering.cpp

```diff
@@ -277,6 +277,96 @@ class LowerBroadcastOp : public OpRewritePattern<mhlo::BroadcastOp> {
   }
 };
 
+// Lower mhlo::SortOp to an pseudo SortOp in the VMLA dialect. This
+// pseudo op generates a set of ordered indices for that array along the last
+// dimension. Then using a torch_index_select the values can be reordered to
+// support arbitrary inputs.
+//
+// TODO(suderman): This lowering only covers the case of ascending values, we
+// should support a separate descending value case by having separate
+// SortAscending and SortDescending operations.
+class LowerSortOp : public OpRewritePattern<mhlo::SortOp> {
+ public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(mhlo::SortOp op,
+                                PatternRewriter &rewriter) const override {
+    auto operand_ty = op.getOperand(0).getType().cast<RankedTensorType>();
+    bool last_dimension = (op.dimension() == -1) ||
+                          (op.dimension() == (operand_ty.getRank() - 1));
+
+    // TODO(suderman): Add transpose to sort along the last dimension.
+    if (!last_dimension) return failure();
+
+    auto &comparator = op.comparator();
+    auto &block = comparator.getBlocks().front();
+    auto &operations = block.getOperations();
+    auto comparison = dyn_cast_or_null<mhlo::CompareOp>(&operations.front());
+
+    // First verify that the block is purely a return of a comparison. This
+    // handles the regular sorting behavior.
+    if (!comparison) return failure();
+
+    auto second = &(*(++operations.begin()));
```

Comment:
nit: could use more descriptive variable name here (second _what_?)

Reply:
Renamed to secondOp.